modules/kernel/src/main/java/org/opencastproject/kernel/security/LtiLaunchAuthenticationHandler.java

```diff
@@ -126,7 +115,7 @@ public LtiLaunchAuthenticationHandler(UserDetailsService userDetailsService, Sec
    * @param untrustedUsersPattern
    */
   public LtiLaunchAuthenticationHandler(UserDetailsService userDetailsService, SecurityService securityService,
-          List<String> highlyTrustedkeys, String untrustedUsersPattern) {
+          List<String> highlyTrustedkeys, String untrustedUsersPattern = "^(admin|opencast_system_account)$") {
```

Comment:
This is not possible in Java. You have to update the three argument constructor, which calls this one and pass `"^(admin|opencast_system_account)$"` instead of `null` as the fourth argument.

Reply:
Also these names are configurable, so this default regexp should probably be constructed from the property values of org.opencastproject.security.admin.user and org.opencastproject.security.digest.user

Reply:
@smarquard I saw that you can get these properties from a `ComponentContext` object via `getProperty` or `OsgiUtil.getComponentContextProperty` but how do I get the `ComponentContext` object? I noticed it is passed to some objects via the `activate` method which is apparently an OSGi callback. But the `LtiLaunchAuthenticationHandler` class doesn't have such a method. Can you tell me how I can add it? Or am I on the wrong track?

Reply:
Is this possible since the `LtiLaunchAuthenticationHandler` is instantiated by Spring and not OSGi?

Changing the default user name is a good point @smarquard. There probably should be a comment in `etc/custom.properties` that reminds users to also update the regex if they use LTI.  I would further suggest to add a note/section to the LTI documentation describing the purpose of the regex. I will create a separat PR for this.